src/client/client.ts

```diff
@@ -275,6 +281,65 @@ export class Client {
     return res;
   }
 
+  /**
+   * Batch publish messages to QStash.
+   */
+  public async batch(
+    req: PublishRequest[]
+  ): Promise<(PublishToUrlResponse | PublishToTopicResponse)[]> {
+    const messages = [];
+    for (const message of req) {
+      const headers = this.processHeaders(message);
```

Comment:
Why don't we do this? It feels like we constructing the same headers without appending anything to them?

```
const headers = await this.processHeaders(message.headers);
      
      messages.push({
        destination: message.url ?? message.topic,
        headers,
        body: message.body,
      });
```

Reply:
The headers that are returned from the function are of type `Headers` and they need to be an object in the body. I tried to `Object.fromEntries` but our version of TS is too old 😅

Reply:
I just double checked and code works just fine 😂. Maybe your editor trying to use older version of TS instead of the one in the package.json? Because our version is `"typescript": "^5.2.2"`